library/bignum.c

```diff
@@ -2243,10 +2243,10 @@ int mbedtls_mpi_exp_mod( mbedtls_mpi *X, const mbedtls_mpi *A,
         /*
          * skip leading 0s
          */
-        if( ei == 0 && state == 0 )
+        if( state == 0 && ei == 0 )
```

Comment:
It is not at all obvious to the average maintainer of Mbed TLS that the order of these tests matters. Please add a comment with a brief explanation and a reference to https://github.com/ARMmbed/mbedtls/issues/4231.